open-api/rest-catalog-open-api.yaml

```diff
@@ -2367,6 +2368,21 @@ components:
           items:
             type: string
 
+    AppendFilesUpdate:
+      allOf:
+        - $ref: '#/components/schemas/BaseUpdate'
+      required:
+        - appended-manifests
+      properties:
+        action:
+          type: string
+          enum: [ "append-files" ]
+        appended-manifests:
```

Comment:
I think we did not really finalize in the discussion threads if this is a good idea. I think it's an easy integration with existing JVM-based engines since they already know how to produce manifests, but if our goal is to make it easy for other non-JVM engines to integrate, seems like this should be a list of DataFiles.

Reply:
Hey Jack, you make a good point. The intentions behind the appended-manifest is to showcase the potential need for compression, and it was a middle ground between a List of DataFiles and a ManifestList. In the case of a large scale Append operation this list can become very large therefore, we can trade some overhead for these operations. However, at the end of the day this is a list of string locations.  How the service appends these files  is depending on it's ability to read [DataFiles](https://iceberg.apache.org/javadoc/master/org/apache/iceberg/DataFiles.html) into their service from the Request body.

Some other potential approaches we could take are:
* Pagination/batching
* Compression (GZIP, Brotli)
* File Upload of a different type other that parquet.

Reply:
I added a similar comment about what content to add below, but I think it's best to choose the lowest common denominator (`ContentFile`/`data_file`) since that would reduce the need to rewrite manifests to properly define things like the sequence number and alleviates the need for a client to know how to construct a manifest.